sonar-php-plugin/src/main/java/org/sonar/plugins/php/PhpExclusionsFileFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.php;
+
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.InputFileFilter;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.utils.WildcardPattern;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.php.api.Php;
+
+public class PhpExclusionsFileFilter implements InputFileFilter {
+
+  private final Configuration configuration;
+
+  private static final Logger LOG = Loggers.get(PhpExclusionsFileFilter.class);
+
+  public PhpExclusionsFileFilter(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  @Override
+  public boolean accept(InputFile inputFile) {
+    if (!Php.KEY.equals(inputFile.language())) {
+      return true;
+    }
+
+    String property = PhpPlugin.PHP_EXCLUSIONS_KEY;
+    String[] excludedPatterns = this.configuration.getStringArray(property);
```

Comment:
I believe this should be stored in a field and not called for every input file.
```
if(excludedPatterns == null) {
    excludedPatterns = WildcardPattern.create(configuration.getStringArray(property))

}
```

And unless you have something in mind I believe the constant could be inlined where it is used rather than having an intermediary variable "property".